src/data_snack/snack.py

```diff
@@ -13,66 +14,113 @@ def _build_key(type_name: Text, *keys: Text):
 
 @dataclass
 class Snack:
+    """
+    A core interface handling saving and reading Entities from db.
+    """
+
     connection: Connection
     registry: Dict[Text, EntityRegistry] = field(default_factory=dict)
 
-    def register_entity(self, cls: EntityType, keys: List[Text], serializer: Serializer = None):
+    def register_entity(self, cls: Type[Entity], key_fields: List[Text], serializer: Serializer = None) -> None:
+        """
+        Registers new Entity type to Snack.
+
+        :param cls: Entity type
+        :param key_fields: a list of fields that will be used to define Entity key
+        :param serializer: Serializer that is used to compress and decompress entities before saving in db
+        """
         type_name = cls.__name__
 
         if type_name in self.registry:
-            raise EntityAlreadyRegistered(f"Entity {type_name} is already registred")
+            raise EntityAlreadyRegistered(f"Entity {type_name} is already registered")
 
-        if not keys:
+        if not key_fields:
             raise WrongKeyValue("Entity keys cannot be empty")
 
-        for key in keys:
+        for key in key_fields:
             if key not in cls.__dataclass_fields__.keys():
                 raise WrongKeyValue(f"Key {key} not found in entity definition.")
 
         if not serializer:
             serializer = DataclassSerializer(cls)
 
         self.registry[type_name] = EntityRegistry(
```

Comment:
Should we allow to remove entity or update the entity? if we want to change key_fields it is not possible now.

Reply:
If you want to update an entity just save the updated object with the same key values.
Removing an entity is good idea, we should cover that.